test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,978 @@
+    enum TestEnum {
+      value1 = 'value1',
+      value2 = 'value2',
+    }
+    enum FailingTestEnum {
+      value3 = 'value3',
+    }
+    const values = Object.values(TestEnum);
+
+    class Bar extends Model<InferAttributes<Bar>, InferCreationAttributes<Bar>> {
+      declare field: TestEnum | FailingTestEnum;
+    }
+
+    Bar.init({
+      field: {
+        type: DataTypes.ENUM(values),
+        validate: {
+          isIn: [values],
+        },
+      },
+    }, {
+      sequelize,
+    });
+
+    const failingBar = Bar.build({ field: FailingTestEnum.value3 });
+
+    await expect(failingBar.validate({ skip: ['field'] })).not.to.be.rejected;
+  });
+
+  it('skips validations for fields with value that is SequelizeMethod', async () => {
+    enum TestEnum {
+      value1 = 'value1',
+      value2 = 'value2',
+    }
+    enum FailingTestEnum {
+      value3 = 'value3',
+    }
+    const values = Object.values(TestEnum);
+
+    class Bar extends Model<InferAttributes<Bar>, InferCreationAttributes<Bar>> {
+      declare field: TestEnum | FailingTestEnum | Utils.Literal;
+    }
+
+    Bar.init({
+      field: {
+        type: DataTypes.ENUM(values),
+        validate: {
+          isIn: [values],
+        },
+      },
+    }, {
+      sequelize,
+    });
+
+    const failingBar = Bar.build({ field: sequelize.literal('5 + 1') });
+
+    await expect(failingBar.validate()).not.to.be.rejected;
+  });
+
+  it('raises an error if saving a different value into an immutable field', async () => {
+    class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+      declare name: string;
+    }
+
+    User.init({
+      name: {
+        type: DataTypes.STRING,
+        validate: {
+          isImmutable: true,
+        },
+      },
+    }, { sequelize });
+
+    await User.sync({ force: true });
+    const user = await User.create({ name: 'RedCat' });
+    expect(user.getDataValue('name')).to.equal('RedCat');
+    user.setDataValue('name', 'YellowCat');
+    const errors = await expect(user.save()).to.be.rejected;
+    expect(errors.get('name')[0].message).to.eql('Validation isImmutable on name failed');
+  });
+
+  it('allows setting an immutable field if the record is unsaved', async () => {
+    class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+      declare name: string;
+    }
+
+    User.init({
+      name: {
+        type: DataTypes.STRING,
+        validate: {
+          isImmutable: true,
+        },
+      },
+    }, { sequelize });
+
+    const user = User.build({ name: 'RedCat' });
+    expect(user.getDataValue('name')).to.equal('RedCat');
+
+    user.setDataValue('name', 'OrangeCat');
+    await expect(user.validate()).not.to.be.rejected;
+  });
+
+  it('raises an error for array on a STRING', async () => {
+    class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+      declare email: string | string[];
```

Comment:
Same as above. The model should be properly typed, but inserting an invalid input in it should be commented with ts-expect-error to let TypeScript know an error should occur there

```suggestion
      declare email: string;
```

Same comment for the following tests